src/cp/index.ts

```diff
@@ -54,9 +58,12 @@ export default class ChargePoint {
     private readonly csUrl: string
   ) { }
 
-  async connect(): Promise<Connection<CentralSystemAction<'v1.6-json'>>> {
+  async connect(connectArgs?: ConnectArgs): Promise<Connection<CentralSystemAction<'v1.6-json'>>> {
     const url = `${this.csUrl}/${this.id}`;
-    const socket = new WebSocket(url, SUPPORTED_PROTOCOLS);
+    const socket = new WebSocket(url, SUPPORTED_PROTOCOLS, {
+      auth: `${this.id}:${connectArgs?.auth}`,
```

Comment:
this always sets the auth key, which may not be desirable for other users of this library. I suggest moving this to the layer above, and just pass the `connectArgs` to the `WebSocket` constructor, that way it won't break any existing code